test/e2e/debug_shared.go

```diff
@@ -163,6 +163,48 @@ func createScDeployRevertedSignedTx(t *testing.T, ctx context.Context, auth *bin
 	return auth.Signer(auth.From, tx)
 }
 
+func createScDeployOutOfGasSignedTx(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client, customData map[string]interface{}) (*ethTypes.Transaction, error) {
+	nonce, err := client.PendingNonceAt(ctx, auth.From)
+	require.NoError(t, err)
+
+	gasPrice, err := client.SuggestGasPrice(ctx)
+	require.NoError(t, err)
+
+	scByteCode, err := testutils.ReadBytecode("ConstructorMap/ConstructorMap.bin")
+	require.NoError(t, err)
+	data := common.Hex2Bytes(scByteCode)
+
+	tx := ethTypes.NewTx(&ethTypes.LegacyTx{
+		Nonce:    nonce,
+		GasPrice: gasPrice,
+		Gas:      uint64(2000000),
+		Data:     data,
+	})
+
+	return auth.Signer(auth.From, tx)
+}
+
+// func createScCreationCodeStorageOutOfGasSignedTx(t *testing.T, ctx context.Context, auth *bind.TransactOpts, client *ethclient.Client, customData map[string]interface{}) (*ethTypes.Transaction, error) {
+// 	nonce, err := client.PendingNonceAt(ctx, auth.From)
+// 	require.NoError(t, err)
+
+// 	gasPrice, err := client.SuggestGasPrice(ctx)
+// 	require.NoError(t, err)
+
+// 	scByteCode, err := testutils.ReadBytecode("FFFFFFFF/FFFFFFFF.bin")
+// 	require.NoError(t, err)
+// 	data := common.Hex2Bytes(scByteCode)
+
+// 	tx := ethTypes.NewTx(&ethTypes.LegacyTx{
+// 		Nonce:    nonce,
+// 		GasPrice: gasPrice,
+// 		Gas:      uint64(150000),
+// 		Data:     data,
+// 	})
+
+// 	return auth.Signer(auth.From, tx)
+// }
+
```

Comment:
Why do we need this commented code?

Reply:
This is the test I've mentioned in the weekly that we need to investigate with the Executor team.
This is commented at the moment to not break the E3E tests and it is going to be used by this issue: #2641